### PR TITLE
Add Debian 12 Bookworm

### PR DESCRIPTION
**Pull request by mayrstefan**
## What does this PR change?

This is some basic preparation to support Debian 12 Bookworm

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- Documentation issue was created: [#2353](https://github.com/uyuni-project/uyuni-docs/issues/2353).

- [x] **DONE**

## Test coverage
- Cucumber tests were added

- [x] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: susemanager/src/mgr_bootstrap_data.py
Comment (srbarrios) [inline susemanager/src/mgr_bootstrap_data.py:1499]: I guess these TODO needs to be replaced before merging the PR?

Comment (mayrstefan) [reply susemanager/src/mgr_bootstrap_data.py:1499]: Yes, someone needs to allocate free IDs

Comment (mcalmer) [reply susemanager/src/mgr_bootstrap_data.py:1499]: This is not just free IDs. We need to setup something for it. Please remove this from this PR.
We need to add it when we have that thing done. It is not directly required for Uyuni. This is more used in SUSE Manager

Comment (mayrstefan) [reply susemanager/src/mgr_bootstrap_data.py]: Removed that section

**Comment by mayrstefan** [comment]
I messed up my git rebase. Hope this is  good enough now

File: testsuite/features/support/constants.rb
Comment (vzhestkov) [inline testsuite/features/support/constants.rb:51]: Why Ubuntu 18.04, which is in EoL already was appended here?

Comment (mayrstefan) [reply testsuite/features/support/constants.rb]: This was not intentional. I guess this was lost in my fight with git merge

File: utils/spacewalk-common-channels.ini
Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:3600]: backports seems to be missing (I see it at https://github.com/uyuni-project/uyuni/pull/7269/files=
```suggestion

[debian-12-amd64-main-backports-uyuni]
label    = debian-12-amd64-main-backports-uyuni
name     = Debian 12 (bookworm) AMD64 Main Backports for Uyuni
archs    = amd64-deb
repo_type = deb
checksum = sha256
base_channels = debian-12-pool-amd64-uyuni
repo_url = http://deb.debian.org/debian/dists/bookworm-backports/main/binary-amd64/
```

**Review by juliogonzalez — APPROVED** [review]
LGTM, however I think the other reviewers should approve in case I am missing something on the testsuite or Java code.

**Review by juliogonzalez — APPROVED** [review]

**Comment by juliogonzalez** [comment]
Reviewers: I asked for a review from you again, as neither our you approved. You left comments, or directly requested changes.

**Review by vzhestkov — APPROVED** [review]
Looks good for me now. @mayrstefan thank you for taking care of it and for removing the change which was not related :+1:

**Review by mackdk — APPROVED** [review]
Java part LGTM. The test failures are due to https://github.com/uyuni-project/uyuni/pull/7604 and are not related to this change itself

**Comment by mackdk** [comment]
I rebase the branch on top of master and clean up the history a little bit. Acceptance tests should now be passing. @mcalmer could you please review again. Seems your requested change is still blocking the merge

**Review by mcalmer — APPROVED** [review]

**Pull request merged**
